server/broadcast.go

```diff
@@ -116,12 +160,24 @@ func (bsm *BroadcastSessionsManager) completeSession(sess *BroadcastSession) {
 	defer bsm.sessLock.Unlock()
 
 	if existingSess, ok := bsm.sessMap[sess.OrchestratorInfo.Transcoder]; ok {
+		if bsm.lastSess != nil && bsm.lastSess.OrchestratorInfo.Transcoder == sess.OrchestratorInfo.Transcoder {
+			if len(bsm.lastSess.SegsInFlight) > 1 {
+				bsm.lastSess.SegsInFlight = bsm.lastSess.SegsInFlight[1:]
+				// don't place session back into the selector in case a segment is still in flight
+				glog.Warningf("NOT COMPLETING SESSION AS SEGMENT IN FLIGHT")
+				return
+			} else {
```

Comment:
If we're returning in the if block, it's common to skip the else block in Go i.e.

```golang
if (len(bsm.lastSess.SegsInFlight) > 1 {
    ...
    return
}

...
bsm.lastSess = sess
```